hummingbot/connector/exchange/beaxy/beaxy_api_order_book_data_source.py

```diff
@@ -42,25 +41,62 @@ def logger(cls) -> HummingbotLogger:
     def __init__(self, trading_pairs: List[str]):
         super().__init__(trading_pairs)
 
+    @classmethod
+    async def init_trading_pair_symbols(cls, shared_session: Optional[aiohttp.ClientSession] = None):
+        """Initialize _trading_pair_symbol_map class variable
+        """
+        client_session = shared_session or aiohttp.ClientSession()
+        async with client_session as client:
+            async with client.get(BeaxyConstants.PublicApi.SYMBOLS_URL, timeout=5) as response:
+                if response.status == 200:
+                    symbols: List[Dict[str, Any]] = await response.json()
+                    cls._trading_pair_symbol_map = {
+                        symbol_data["name"]: (f"{symbol_data['baseCurrency']}-{symbol_data['termCurrency']}")
+                        for symbol_data in symbols
+                    }
+                else:
+                    IOError(f"There was an error requesting the list of symbols from the exchange ({response})")
+
+    @classmethod
+    async def trading_pair_symbol_map(cls) -> Dict[str, str]:
+        if not cls._trading_pair_symbol_map:
+            await cls.init_trading_pair_symbols()
+
+        return cls._trading_pair_symbol_map
+
+    @staticmethod
+    async def exchange_symbol_associated_to_pair(trading_pair: str) -> str:
+        symbol_map = await BeaxyAPIOrderBookDataSource.trading_pair_symbol_map()
+        symbols = [symbol for symbol, pair in symbol_map.items() if pair == trading_pair]
+
+        if symbols:
+            symbol = symbols[0]
+        else:
+            raise ValueError(f"There is no symbol mapping for trading pair {trading_pair}")
+
+        return symbol
+
+    @staticmethod
+    async def trading_pair_associated_to_exchange_symbol(symbol: str) -> str:
+        symbol_map = await BeaxyAPIOrderBookDataSource.trading_pair_symbol_map()
+        return symbol_map[symbol]
+
     @staticmethod
     async def fetch_trading_pairs() -> Optional[List[str]]:
```

Comment:
I see that a two-way map won't work here. Maybe maintaining a forward and inverse map?